internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -354,6 +415,184 @@ func setCPUQuota(cpuMountPoint, parentDir string, c *oci.Container, enable bool)
 	return nil
 }
 
+// setCPUPMQOSResumeLatency sets the pm_qos_resume_latency_us for a cpu and stores the original
+// value so it can be restored later. If the latency is an empty string, the original latency
+// value is restored.
+func setCPUPMQOSResumeLatency(c *oci.Container, latency string) error {
+	return doSetCPUPMQOSResumeLatency(c, latency, sysCPUDir, sysCPUSaveDir)
+}
+
+// doSetCPUPMQOSResumeLatency facilitates unit testing by allowing the directories to be specified as parameters.
+func doSetCPUPMQOSResumeLatency(c *oci.Container, latency, cpuDir, cpuSaveDir string) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return fmt.Errorf("find container %s CPUs", c.ID())
+	}
+
+	cpus, err := cpuset.Parse(lspec.Resources.CPU.Cpus)
+	if err != nil {
+		return err
+	}
+
+	for _, cpu := range cpus.ToSlice() {
+		latencyFile := fmt.Sprintf("%s/cpu%d/power/pm_qos_resume_latency_us", cpuDir, cpu)
+		cpuPowerSaveDir := fmt.Sprintf("%s/cpu%d/power", cpuSaveDir, cpu)
+		latencyFileOrig := path.Join(cpuPowerSaveDir, "pm_qos_resume_latency_us")
+
+		if latency != "" {
+			// Retrieve the current latency.
+			latencyOrig, err := os.ReadFile(latencyFile)
+			if err != nil {
+				return err
+			}
+
+			// Save the current latency so we can restore it later.
+			err = os.MkdirAll(cpuPowerSaveDir, 0o750)
+			if err != nil {
+				return err
+			}
+			err = os.WriteFile(latencyFileOrig, latencyOrig, 0o644)
+			if err != nil {
+				return err
+			}
+
+			// Update the pm_qos_resume_latency_us.
+			err = os.WriteFile(latencyFile, []byte(latency), 0o644)
+			if err != nil {
+				return err
+			}
+		} else {
```

Comment:
nit: can we continue here instead of nesting the rest with `else {`

Reply:
Judging by your comment below - I assume this isn't really a "nit" because you are asking me to change this before it gets merged. I actually think it's more readable the way it is but that could be because I'm new to golang. I guess the golang preference is to avoid nesting where possible and return (or continue) early? I can update this, but won't be until next week as tomorrow is a holiday here in Canada.

Reply:
yeah in general I prefer flows with the least indentation. Thank you, no rush on my end :+1:

Reply:
Done